lib/tapioca/dsl/compilers/active_job.rb

```diff
@@ -83,6 +83,7 @@ def decorate
         end
         def perform_later_parameters(parameters, constant_name)
           if ::Gem::Requirement.new(">= 7.0").satisfied_by?(::ActiveJob.gem_version)
+            parameters.reject! { |typed_param| typed_param.param.name == "block" }
```

Comment:
There is no guarantee that the original block parameter will be called `"block"`, so this won't work in the general case.